Worker/src/Common/Archive/ZipArchiver.cs

```diff
@@ -33,113 +33,27 @@
 using ArmoniK.DevelopmentKit.Common;
 using ArmoniK.DevelopmentKit.Common.Exceptions;
 
-namespace ArmoniK.DevelopmentKit.Worker.Common;
+namespace ArmoniK.DevelopmentKit.Worker.Common.Archive;
 
-public class ZipArchiver
+public class ZipArchiver : IArchiver
 {
-  private static readonly string RootAppPath = "/tmp/packages";
-
-  /// <summary>
-  /// </summary>
-  /// <param name="assemblyNameFilePath"></param>
-  /// <returns></returns>
-  public static bool IsZipFile(string assemblyNameFilePath)
-  {
-    //ATm ONLY Check the extensions 
-
-    var extension = Path.GetExtension(assemblyNameFilePath);
-    if (extension?.ToLower() == ".zip")
-    {
-      return true;
-    }
-
-    return false;
-  }
-
-  /// <summary>
-  /// </summary>
-  /// <param name="assemblyNameFilePath"></param>
-  /// <returns></returns>
-  /// <exception cref="WorkerApiException"></exception>
-  public static IEnumerable<string> ExtractNameAndVersion(string assemblyNameFilePath)
+  private const           string      RootAppPath = "/tmp/packages";
+  private static readonly ZipArchiver Instance    = new();
```

Comment:
Why do we have a Singleton here?

Reply:
Because static interface methods are only coming in c#11...